src/backend/commands/createas.c

```diff
@@ -363,6 +393,22 @@ ExecCreateTableAs(ParseState *pstate, CreateTableAsStmt *stmt,
 		save_nestlevel = NewGUCNestLevel();
 	}
 
+	if (is_matview && into->ivm)
+	{
+		/* check if the query is supported in IMMV definition */
+		if (contain_mutable_functions((Node *) query))
+			ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("mutable function is not supported on incrementally maintainable materialized view"),
+					 errhint("functions must be marked IMMUTABLE")));
+
+		check_ivm_restriction((Node *) query);
+
+		/* For IMMV, we need to rewrite matview query */
+		query = rewriteQueryForIMMV(query, into->colNames);
+		query_immv = copyObject(query);
+	}
```

Comment:
New variable `query_immv` make us lost some logic like `rewritten = QueryRewrite(query);` as below which IMV patch28 didn't do that.

Reply:
query_immv is just used to create triggers for materialized view.

Reply:
Why avoid rewrite?

Reply:
we just need base tables , not include rewrite table from rule.